accepted/automatic-redirects-by-default/automatic-redirects-by-default.md

```diff
@@ -0,0 +1,161 @@
+to turn the setting off if necessary. In other words, we change it from an opt-
+in feature to an opt-out feature.
+
+[binding redirects]: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/redirect-assembly-versions
+[abrg]: https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/how-to-enable-and-disable-automatic-binding-redirection
+
+## Scenarios and User Experience
+
+### Consuming a .NET Standard 2.0 library from .NET Framework 4.6.1
+
+Michelle maintains a .NET Framework 4.5 application. She tries to install the
+new `System.IO.Pipelines` NuGet package that targets .NET Standard 2.0. She gets
+an error message from NuGet that this isn't compatible with .NET Framework 4.5.
+After some research, she learns that she must at least target .NET Framework
+4.6.1 in order to use .NET Standard 2.0 packages. After upgrading to .NET
+Framework 4.6.1, the package installs without errors. Michelle then starts to
+use pipelines from her app. When she F5s her application it just works fine.
+
+Michelle doesn't know that this is thanks to automatic binding redirect
+generation being on by default. Otherwise her application would have crashed
+with a `FileLoadException`.
+
+### Upgrading a NuGet dependency
+
+Robert is using FizzBuzz, a documentation framework that produces help files out
+of Markdown documents. FizzBuzz depends on Markdig, a popular .NET markdown
+library. Robert encounters an issue with some of his recent documentation
+changes which causes Markdig to crash. He notices that while there is no updated
+version of FizzBuzz, there is an updated version of Markdig that fixes his
+issue. Robert updates his (indirect) dependency on Markdig to the latest version
+that fixes the issue. He triggers a rebuild of the documentation and is happy to
+see that the crash no longer occurs.
+
+Robert doesn't know that this is thanks to automatic binding redirect generation
+being on by default. Otherwise his application would have crashed with a
+`FileLoadException`.
+
+## Requirements
+
+### Goals
+
+* **Turning this setting on must not break working applications**. We expect
+  this setting to make more applications work, but we must not break
+  applications where adding binding redirects breaks them.
+
+* **It must be possible to opt-out**. The customer base on .NET Framework is
+  fairly large and has an unfathomable amount of different configurations. We
+  need to assume that this setting might not work for some extremely rare
+  condition and thus allow those customers to opt-out.
+
+### Non-Goals
+
+* **Introducing new binding policy**. This proposal is only about enabling an
+  already existing feature in MSBuild which uses the also existing binding
+  redirect feature of the CLR.
+
+## Design
+
+Somewhere in the common props we need to add the following snippet:
+
+```xml
+<PropertyGroup>
+    <AutoGenerateBindingRedirects Condition="'$(AutoGenerateBindingRedirects)' == ''">True</AutoGenerateBindingRedirects>
```

Comment:
So the opt-out is for the customer to set this to false in their project?

Reply:
Correct

Reply:
Blindly turning this true everywhere is going to cause perf issues.